src/main/java/io/vertx/core/buffer/impl/VertxByteBufAllocator.java

```diff
@@ -11,22 +11,52 @@
 package io.vertx.core.buffer.impl;
 
 import io.netty.buffer.AbstractByteBufAllocator;
+import io.netty.buffer.AdaptiveByteBufAllocator;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufAllocator;
 import io.netty.buffer.PooledByteBufAllocator;
 import io.netty.buffer.UnpooledByteBufAllocator;
 import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.SystemPropertyUtil;
 
 public abstract class VertxByteBufAllocator extends AbstractByteBufAllocator {
 
   /**
    * Vert.x pooled allocator.
```

Comment:
missing: document how this allocator is chosen

Reply:
I've added some docs: PTAL

What's "weird" is that I'm reusing the existing `-Dio.netty.allocator.type` property to control which allocator use, but currently anything different from `adaptive` will fallback to `PooledByteBufAllocator.DEFAULT` because that's the sole choice for `VertxByteBufAllocator.POOLED_ALLOCATOR`, which, as the name suggest, must be pooled.

Probably it's worthy to use a separate vertx sys property here, for this, but TBH I would expect if the the Netty one is using `adaptive` to be able to reuse the instance allocated by Netty at `ByteBufAllocator.DEFAULT`.